atlas/packages/components/src/components/VideoPreview/VideoPreview.tsx

```diff
@@ -9,58 +23,90 @@ type VideoPreviewProps = {
 	showChannel: boolean;
 	showMeta: boolean;
 	createdAt: string;
+	duration?: string;
+	// video watch progress in percent (0-100)
+	progress?: number;
 	views: string;
 	poster: string;
-	onClick: (e: React.MouseEvent<HTMLElement>) => void;
 	imgRef: React.Ref<HTMLImageElement>;
-	onChannelClick: (e: React.MouseEvent<HTMLElement>) => void;
-} & VideoPreviewStyleProps;
+	onClick?: (e: React.MouseEvent<HTMLElement>) => void;
+	onChannelClick?: (e: React.MouseEvent<HTMLElement>) => void;
+};
 
 const VideoPreview: React.FC<Partial<VideoPreviewProps>> = ({
 	title,
 	channel,
 	channelImg,
-	showChannel,
-	showMeta,
+	showChannel = true,
+	showMeta = true,
 	createdAt,
+	duration,
+	progress = 0,
 	views,
 	imgRef,
 	poster,
-	onClick = () => {},
-	onChannelClick = () => {},
-	...styleProps
+	onClick,
+	onChannelClick,
 }) => {
-	let styles = makeStyles({ showChannel, poster, ...styleProps });
+	const clickable = !!onClick;
+	const channelClickable = !!onChannelClick;
+
+	const handleChannelClick = (e: React.MouseEvent<HTMLElement>) => {
+		if (!onChannelClick) {
+			return;
+		}
+		e.stopPropagation();
+		onChannelClick(e);
+	};
+
+	const handleClick = (e: React.MouseEvent<HTMLElement>) => {
```

Comment:
`handleClick` and `handleChannelClick` look almost identical, and the use case of triggering a callback only if some predicate is met looks very common. 
It improves the UX  since if cursor changes, then there is _for sure_ something to click. We could use this in other places where we have `onClick` handlers like buttons and so on.
I suggest to: 
- Define an HOF that takes a Callback, and returns a function which fires the callback iff a condition is met. In our example:

```javascript

const when = predicate => fn => e => predicate ? fn() : undefined
const handleClick = myFn(clickable)(onClick)
const handleChannelClick = myFn(channelClickable)(onChannelClick)
```

- Move this to `utils`.

I've left out calling `e.stopPropagation( )` from above, to make the example smaller, but you get the idea. This might be out of scope, so we could open an issue and move it to a future PR.

Reply:
I'm not sure if this is an abstraction worth having at this time:
- Checking a predicate inside a callback isn't a lot of work, especially once you have an event handler anyway. It's a matter of adding one `if` line
- The specific conditions and order in which you want to check them may be pretty custom to the callback. Handlers may start off pretty simple, allowing the use of the helper, but grow in time, requiring moving the condition inside the handler, making refactor more work
- I believe it would be easier to read when the condition is inside the handler itself, you don't need to check any helper function living outside of the component for its behaviour, you just look at the handler